packages/flutter/lib/src/painting/border_radius.dart

```diff
@@ -395,10 +395,10 @@ class BorderRadius extends BorderRadiusGeometry {
   RRect toRRect(Rect rect) {
     return RRect.fromRectAndCorners(
       rect,
-      topLeft: topLeft,
-      topRight: topRight,
-      bottomLeft: bottomLeft,
-      bottomRight: bottomRight,
+      topLeft: topLeft.clamp(minimum: Radius.zero), // ignore_clamp_double_lint
```

Comment:
Maybe create a `RRect.clampCorners` method? that applies one minimum/maximum radius to all four corners?

Reply:
Sure, I could, but really this won't be a problem once the `RRect` doesn't allow negative values anymore (which will happen shortly).  I mean, sure, clamping to something other that zero might also be important, but it's not something that people have requested.